test/test_producer.py

```diff
@@ -40,3 +41,45 @@ def partitions(topic):
         topic = b"test-topic"
         producer.send_messages(topic, b'hi')
         assert client.send_produce_request.called
+
+    def test_producer_async_queue_overfilled_batch_send(self):
+        queue_size = 2
+        producer = Producer(MagicMock(), batch_send=True, maxsize=queue_size)
+
+        topic = b'test-topic'
+        partition = 0
+        message = b'test-message'
+
+        with self.assertRaises(Full):
+            message_list = [message] * (queue_size + 1)
+            producer.send_messages(topic, partition, *message_list)
+        self.assertEqual(producer.queue.qsize(), queue_size)
+
+    def test_producer_async_queue_overfilled(self):
```

Comment:
you could theoretically decompose the body into a helper method, since the only difference between this test and the one above, `test_producer_async_queue_overfilled_batch_send` is how the `Producer` is initialized.

not a big deal, just something to remember next time